controller/linodemachine_controller.go

```diff
@@ -214,7 +216,7 @@ func (r *LinodeMachineReconciler) reconcile(ctx context.Context, logger logr.Log
 	// Make sure bootstrap data is available and populated.
 	if machineScope.Machine.Spec.Bootstrap.DataSecretName == nil {
 		logger.Info("Bootstrap data secret is not yet available")
-		conditions.MarkFalse(machineScope.LinodeMachine, ConditionPreflightMetadataSupportConfigured, WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
+		conditions.MarkFalse(machineScope.LinodeMachine, ConditionPreflightLinodeFirewallReady, WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
I think we still want to leave this line as is?

Reply:
We don't have any condition for bootstrap data secret. Here, we are returning the first condition we are going to hit. Previously, it was `ConditionPreflightCreated`. When we added step to calculate whether to use cloud-init or stackscripts, we changed it to first condition which was `ConditionPreflightMetadataSupportConfigured `. Since now we are checking if firewall condition is successful or not, we are changing it to that first condition. Do you think we should add another condition here for bootstrap secret and return accordingly?

Reply:
We do have a condition for the bootstrap data since it takes a few seconds to create. I don't think it really matters which one comes first since they are both needed.

Reply:
Added new condition for this. Showing `ConditionPreflightMetadataSupportConfigured ` or `ConditionPreflightLinodeFirewallReady` with reason `WaitingForBootstrapData` in status is a bit confusing, hence thought we should have its own condition `ConditionPreflightBootstrapDataSecretReady`